packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.d.ts

```diff
@@ -27,7 +27,7 @@ export function createFilterOptions<T>(
 
 export type AutocompleteFreeSoloValueMapping<FreeSolo> = FreeSolo extends true ? string : never;
 
-export type AutocompleteValue<T, Multiple, DisableClearable, FreeSolo> = Multiple extends true
+export type AutocompleteValue<T, Multiple, DisableClearable, FreeSolo> = true extends Multiple
```

Comment:
Can you explain why you inverted it and why is it now constrained to `Multiple` type?

Reply:
Sorry for replying so late...my family and I got covid these days. 

I inverted it because we can ensure the inferred type is `Array<T | AutocompleteFreeSoloValueMapping<FreeSolo>>` only if the multiple prop is `true`. And I explained [here](https://github.com/mui/material-ui/issues/35308#issuecomment-1338729629) why `Multiple extends true` does not work.

Reply:
I think the root fix would be to update the default types:
```diff
--- a/packages/mui-material/src/Autocomplete/Autocomplete.d.ts
+++ b/packages/mui-material/src/Autocomplete/Autocomplete.d.ts
@@ -287,7 +287,7 @@ export interface AutocompleteProps<
  */
 export default function Autocomplete<
   T,
-  Multiple extends boolean | undefined = undefined,
-  DisableClearable extends boolean | undefined = undefined,
-  FreeSolo extends boolean | undefined = undefined,
+  Multiple extends boolean | undefined = false,
+  DisableClearable extends boolean | undefined = false,
+  FreeSolo extends boolean | undefined = false,
 >(props: AutocompleteProps<T, Multiple, DisableClearable, FreeSolo>): JSX.Element;
```

because in the implementation the props are `false` by default. See https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Autocomplete/Autocomplete.js#L407. 
That way the `Multiple` parameter type will be `false` if multiple is not defined and there would be no need to invert the constraint then.

Can you check if it works?

Reply:
Make sense! I've tested this, and it works very well and looks like a more reasonable way to fix this issue. So I'll change my solution to this. Thanks!